executor/builder.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/column"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/optimizer/plan"
+	"github.com/pingcap/tidb/parser/opcode"
+	"github.com/pingcap/tidb/sessionctx/autocommit"
+)
+
+// executorBuilder builds an Executor from a Plan.
+// the InfoSchema must be the same one used in InfoBinder.
+type executorBuilder struct {
+	ctx context.Context
+	is  infoschema.InfoSchema
+}
+
+func newExecutorBuilder(ctx context.Context, is infoschema.InfoSchema) *executorBuilder {
+	return &executorBuilder{
+		ctx: ctx,
+		is:  is,
+	}
+}
+
+func (b *executorBuilder) build(p plan.Plan) Executor {
+	switch v := p.(type) {
+	case *plan.TableScan:
+		return b.buildTableScan(v)
+	case *plan.IndexScan:
+		return b.buildIndexScan(v)
+	case *plan.Filter:
+		return b.buildFilter(v)
+	case *plan.SelectLock:
+		return b.buildSelectLock(v)
+	case *plan.SelectFields:
+		return b.buildSelectFields(v)
+	case *plan.Sort:
+		return b.buildSort(v)
+	case *plan.Limit:
+		return b.buildLimit(v)
+	}
```

Comment:
maybe add a log or assert here is better, if build() returns nil, it will cause a crash.